### PR TITLE
DISP-3971: Add ability to specify analytics tag

### PR DESCRIPTION
**Pull request by derekcaneja**
A transaction ID will be created for each authenticated request.
Both the transaction ID and analytics tags will be added to the request headers.

### CODE REVIEW

File: lib/dispatch.js [hunk 1]
```diff
@@ -274,6 +275,23 @@ export default class Dispatch {
   }
 
   /**
+   * Append analyitcs tags and a transaction ID to the headers.
+   * @param  {[type]} options A set of options for a request.
+   * @return {Object}         The new set of options.
+   */
+  appendAnalyticsHeaders(options) {
+    const analyticsTags = _.map(_.pairs(options.analytics), param => param.join(':')).join(',');
+    const transactionId = UUID.v4();
```
Comment (jraede) [inline lib/dispatch.js]: I'd allow people to provide a custom transaction ID in the options so we can generate it prior to the API call and use in analytics. e.g. `options.transactionID`

File: lib/dispatch.tests.js [hunk 2]
```diff
@@ -198,7 +198,7 @@ describe('client SDK', () => {
       });
     });
 
-    describe('wrapped authenticated request', () => {
+    describe.only('wrapped authenticated request', () => {
```
Comment (jraede) [inline lib/dispatch.tests.js]: Remove `only`

**Comment by aldeed** [comment]
LGTM

**Comment by jraede** [comment]
🐑

**Pull request merged**